### DISCUSSION TITLE
Copy-PnPList: List copy failed with error You do not have sufficient permission to perform this operation.

### DISCUSSION THREAD

**Discussion by diltont** [pnp/powershell#4619] [General]
### Reporting an Issue or Missing Feature
I am trying to create a copy of SharePoint list in the same site and I am getting permission error.

### Expected behavior 
Copy list works fine

### Actual behavior 
When I use the cmdlet "Copy-PnPList -Identity "My List" -Title "Copy of My List", I am getting the follwoing error:
"Copy-PnPList: List copy failed with error You do not have sufficient permission to perform this operation."

### Steps to reproduce behavior

- I imported the PnP module
- Then I connected the SharePoint using the following cmdlet :
"Connect-PnPOnline -Url "https://contoso.sharepoint.com" -ClientId 6c5c98c7-e05a-4a0f-bcfa-0cfc65aa1f28 -Tenant 'contoso.onmicrosoft.com' -Thumbprint 34CFAA860E5FB8C44335A38A097C1E41EEA206AA".
- The connection was successful and I verified it with Get-PnPConnection
- Then I tried two other commands: a)Find-PnPFile -Folder "Shared Documents/Sub Folder" -Match *.docx
b)Get-PnPFile -Url /sites/project/SiteAssets/image.jpg -Path c:\temp -FileName image.jpg -AsFile. Both of these cmdlets worked successfully.
-  Then I tried the cmdlet "Copy-PnPList -Identity "My List" -Title "Copy of My List" and I got the error as mentioned above.
I am the owner of the SharePoint site and for the azure app registration , I have giveen  Sites.Selected permission under application permissions.

### What is the version of the Cmdlet module you are running?
I started with version 2.12. Then I updated to latest version nightly 2.99.65. Both have same problem

### Which operating system/environment are you running PnP PowerShell on?
- Windows

**Comment by waaromikniet** [comment]
Did you run this cmdlet: https://pnp.github.io/powershell/cmdlets/Grant-PnPAzureADAppSitePermission.html
To give permissions for the sites.selected?

**Reply by diltont** [reply]
Thanks for the reply. 
I am trying to access the SharePoint through app only access way (non interactive).
I have given Microsoft.Graph.Sites.Selected permissions  under  type application also plus the Sharepoint.sites.selected of type application as well.
Still I cant copy sharepoint lists . 
The following image is from PnP documentation. it says to remove Graph permissions. With and without graph permissions, the problem still exists.
<img width="563" alt="image" src="https://github.com/user-attachments/assets/7ffe5448-e907-409b-b5e6-4e5418ca3204">

Do I need to advanced permissions (Microsoft graph.Sites.FullControl .All and SharepointSites.FullControl .All) to get this done?
But it would be difficult becuase of company policies

**Comment by waaromikniet** [comment]
Adding the sites.Selected application permissions is not enough. You have to configure the site(s) and permission for the app. The command: "Grant-PnPAzureADAppSitePermission" does this. I think you need full control to be able to copy a list. Full control is then only added for the configured site. So no need for the application permission Microsoft graph.Sites.FullControl .All and SharepointSites.FullControl .All
